### PR TITLE
Code Smells

### PR DESCRIPTION
**Pull request by Peaceful36**

### CODE REVIEW

**Comment by Peaceful36** [comment]
Identifying different types of Code Smells

**Review by auroraborie — COMMENTED** [review]
Hi there, good job on your article! I found it informative and a good reminder for my own coding practices. I had a few suggestions for your PR to make things clearer for the future students reading them as follows:

- There are a few formatting discrepancies in your file where sentences cut off to the next line. On line 5 and 40 I noticed in particular. 
- Because the reader would have to scroll to through the entire article to see all the code smells you mentioned, I recommend a table of contents in the first section to help organize your information and let potential readers know what they're going to learn about beforehand 
- In each code smell section there is one code example of that code smell which is helpful, but it would be great if we could see some more examples (through screenshots or videos if code takes up too much space).
-  It could also be beneficial to link to an external website for readers to learn about how to fix these code smells as a logical next step, if they want to learn more

**Comment by Peaceful36** [comment]
Hey, thanks for your opinions for my article. Reading through my work again, I find the comments are great to help me improve it more.

**Comment by Arick-mux** [comment]
Hey Hoa, nice descriptions and quite well written! I would like to say that your instructions and explanations are very succinctly and thoroughly written, the examples are also highly intuitive. 

What I would add is some real life examples as well as some more situations in life which code smelling can be predominant, such as a website or a banking app. Of course more videos and pictures would make clearer differences, especially pictures that compare real-life examples so that students are able to understand the consequences if they happen to stumble across similar issues in the future! 

Also, as someone who also got aware of code smells, I would try to maybe organize the different types of code smells in your content by their general types. 

Some of the general types that are pretty common to me would be: (1) unnecessary or meaningless code that can be deleted or replaced; (2) oversized code with long methods, parameters, or classes; (3) Object Oriented Abusers, which is a smelling mistake novices of OOP usually make due to not fully implementing or utilizing all the potential OOP possibilities. and (4) the preventer types which makes future modification of codes or programs not possible or difficult. Of course these are just some general types out there so far, but by implementing all of these you would be able to cover most if not all types of code smells! 

Finally, at the end of your article maybe you can implement a conclusion about the pros and cons of leniency in code smells, and what are some of the easiest solutions to deal with it from a novice's perspective. Keep up the good work!

**Comment by LycheeKingwyz** [comment]
Hi @Peaceful36 , I just read through your file and I really like the topic! Below are some suggestions:

1. Very tiny mistake: missing a bracket in the first paragraph of the refactoring section.
2. It would make the refactoring part more clear if you add some examples to show how to improve the code you mentioned with the technique of refactoring.

And I want to give you a comliment on the choice of your wording, since the target audience of this article would mostly be entry level programmers, it is indeed a good choice to keep the language simple and intuitive. Keep up the good work!

File: Topics/Development_Process/Code Smells/Code_Smells.md
Comment (LycheeKingwyz) [inline Topics/Development_Process/Code Smells/Code_Smells.md:1]: Hi @Peaceful36 , I like the content you wrote. Code smell is an interesting topic, and I really learned something from your article. Below is some suggestion on typesettling:
1. There is one line with extra space in the intro part("what is a code smell?").
2. The hierarchy of table of contents seems confusing and not matching. Maybe what you can do is to change What is a code smell? and Common Types of Code Smells and How They Can Harm your Codes to ## tiltles, and different types ### titles.

File: Topics/Development_Process/Code Smells/Code_Smells.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+4. [Dead Code](#dead-code)
+5. [Middle Man](#middle-man)
+6. [Feature Envy](#feature-envy)
+7. [Long Functions](#long-functions)
+8. [Data Clumps](#data-clumps)
+9. [Additional Resources](#additional-resources)
+
+### What is a code smell?
+
+Code smells are indicators of potential issues or weaknesses in software code. They are not bugs or errors, but rather patterns or
```
Comment (LycheeKingwyz) [inline Topics/Development_Process/Code Smells/Code_Smells.md:16]: the space at end of line should be deleted

File: Topics/Development_Process/Code Smells/Code_Smells.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+8. [Data Clumps](#data-clumps)
+9. [Additional Resources](#additional-resources)
+
+### What is a code smell?
+
+Code smells are indicators of potential issues or weaknesses in software code. They are not bugs or errors, but rather patterns or 
+
+practices that may lead to problems in the future. 
+
+# Common Types of Code Smells and How They Can Harm your Code
```
Comment (LycheeKingwyz) [inline Topics/Development_Process/Code Smells/Code_Smells.md:20]: The typesettling hierarchy here seems confusing, maybe change the title here to a ## title, and ### for all the titles within.

**Review by LycheeKingwyz — CHANGES_REQUESTED** [review]

**Review by Arick-mux — COMMENTED** [review]
From the last visit, the structures are much more organized and the logic is much more fluent! The examples are solid too. The only thing I may be wanting to change is to explain the examples using internal commenting rather than just using paragraphs at the end of each code example in order to walk user through each step. This is to prevent the case that they are new to coding or the examples you have chosen!

**Pull request merged**